### ISSUE TITLE
Logged Out of Keybase CLI, Must Use New Device Name to Log In Again

### ISSUE THREAD

**Issue by Californian** [keybase/client#2463]
Somehow my `keybase` CLI has been logged out (this has happened several times). I then try to log in again and cannot do so without creating a new device name. I have a device naming scheme that I would like to stick to, so coming up with a new name is not preferable (since it is the same machine). I have tried authentication via every method, and then resorted to resetting my account, but none of these steps have resolved the issue.

**Comment by cjb** [comment]
@Californian Please could you run `keybase log send` in case that helps us figure it out?

@patrickxb Looks like a deprovisioning problem, any ideas?

**Comment by cjb** [comment]
@Californian By the way, the restriction on device name reuse is important to have for security reasons, so that's unlikely to go away.  But it's a nasty bug any time your CLI forgets who you are and we'd like to fix that.

**Comment by Californian** [comment]
@cjb yeah I noticed someone mentioned that security restriction in another issue.
Log ID: `2437408838c65f3c5882ed1c`

**Comment by Californian** [comment]
In terms of practical next steps from my end, is it possible I'll be able to authenticate this device as the same one, or will I definitely have to rename it? If it's possible, I can wait for investigation, but if it's not, I'll just go ahead and rename it now.

**Comment by maxtaco** [comment]
Thanks @Californian, definitely something up but our log send was pretty sparse on details.  I think you have an older version of keybase, can you upgrade for us?   We've solved a bunch of bugs since then, but no guarantees that yours will be one of them.

Cc: @oconnor663 and @patrickxb

**Comment by zmanian** [comment]
This sounds a lot like what I was dealing with in keybase/keybase-issues#2106

**Comment by oconnor663** [comment]
It looks like v1.0.14-0 is the latest version available in the official Arch repos. (@Californian are you running Arch Linux?) Felix Yan's packaging jobs must not've pulled 1.0.14-1 for some reason. Not a big deal in this case, since -1 was tagged to fix a Debian-only packaging issue, but I'll find out why.

**Comment by maxtaco** [comment]
@Californian have you been manually deleting files from `~/.config/keybase` by any chance?

**Comment by Californian** [comment]
@oconnor663 Yep, I'm on Arch. I assumed that kind of info was in the logs, but if you need anything else along those lines let me know.
@maxtaco I am on the latest version in community (version `1.0.14.0-1`), but I can move to the `git` version if that might help (version `1.0.14_20160328133147+256dedd-1`).
I haven't manually touched any files in `~/.config/keybase`.
@maxtaco that file (`~/.config/keybase/config.json`) doesn't exist; the parent folder (`~/.config/keybase`) just has a `secretkeys.[username].mpack` file.

**Comment by maxtaco** [comment]
So the issue is a lack of `~/.config/keybase/config.json`, though we don't know why....  One thing to try now might be a `keybase -d login`.  Thanks @Californian

**Comment by fenixnet-net** [comment]
I'm having the same issue.

log send id 8af856d41c9bb10e64baca1c

I'm running 1.0.15 on mac os v10.11.13 via homebrew. I'm having a combo of this issue and https://github.com/keybase/keybase-issues/issues/2106.

I log in and cannot reclaim the device name. Create a new one, verify, client says great.
Then I try to do decrypt and client says I'm not logged in.
Repeat forever, burning device names as I go.

**Comment by maxtaco** [comment]
We'll take a look this week if we can, but we're up against a deadline, so
we might be a little slow in getting to it.


On Sat, Apr 30, 2016 at 1:58 PM, Cameron Nicholson <notifications@github.com

**Comment by fenixnet-net** [comment]
If it's helpful, here's the commands I ran just now with results:

keybase login
result: "✔ Success! You provisioned your device laptop.
You are logged in as jadefalcon1"

keybase decrypt
"▶ ERROR No device found for device ID 17018c9dbb99299e0d9abe6984e56818"

keybase log send
"▶ WARNING error opening log "/Users/cameron/Library/Logs/keybase.kbfs.log": open /Users/cameron/Library/Logs/keybase.kbfs.log: no such file or directory
▶ WARNING error opening log "/Users/cameron/Library/Logs/Keybase.app.log": open /Users/cameron/Library/Logs/Keybase.app.log: no such file or directory"
Log ID afd75fbae42bd134c4d3021c

Also, ~/.config/keybase doesn't exist. I'm guessing that's where things are falling over, since that means my state is evaporating as soon as the login routine exits (assuming I understand the behavior correctly - I might not).

I'll wait for email notification on the thread. Thanks!

**Comment by patrickxb** [comment]
@fenixnet-net could you please paste the contents of 

    /Users/cameron/Library/Application Support/Keybase/config.json

into a comment here?  It seems like there is an issue with a device entry in your config file and it would help to see the whole thing.  `config.json` doesn't contain any private information, so it is ok to post.

Thank you!

**Comment by fenixnet-net** [comment]
Here we go:

`{
    "current_user": "jadefalcon1",
    "users": {
        "jadefalcon1": {
            "device": "17018c9dbb99299e0d9abe6984e56818",
            "id": "ad9b4ad1764c46000f04943d8e884c19",
            "name": "jadefalcon1",
            "salt": "e203102c3527b262208e6e2614574b65"
        }
    }
}`

**Comment by patrickxb** [comment]
@fenixnet-net I believe I've tracked down (and fixed) the bug that is preventing you from using this device in any meaningful way.  

I'm not 100% sure how you got into this state and am looking into that now.  Was there an error that you can recall the first time you provisioned this device?

Also, I can help you manually get out of this state and use this device while waiting for our fix to be released.  What is the name of the device that is currently stuck?  `ufomac` or `laptop`?

Thanks for your patience and helping us track this down.

**Comment by fenixnet-net** [comment]
It's been a while since I used this machine's keybase CLI. I'm sorry I don't have any recollection of any obvious errors or issues.

I believe the current device name is laptop. Ufomac was from my penultimate attempt.

**Comment by patrickxb** [comment]
No problem!

So to get it working now, replace your current `/Users/cameron/Library/Application Support/Keybase/config.json` file with this:

    { 
        "current_user": "jadefalcon1", 
        "users": { 
            "jadefalcon1": { 
                "device": "2dd0891fbc8a3318d13debe30261b718", 
                "id": "ad9b4ad1764c46000f04943d8e884c19", 
                "name": "jadefalcon1", 
                "salt": "e203102c3527b262208e6e2614574b65" 
            }
         }
     }

(this manually sets the correct device ID for you).

And then run `keybase login` and you should be all set.

**Comment by fenixnet-net** [comment]
Very sad news - still busticated.

ufo:~ cameron$ keybase login
▶ WARNING device id mismatch session <-> env
Your keybase username or email address: cameron.nicholson@fenixnet.net
Please enter the Keybase passphrase for cameron.nicholson@fenixnet.net (12+ characters): 
▶ ERROR Bad passphrase: server rejected login attempt.
Please enter the Keybase passphrase for cameron.nicholson@fenixnet.net (12+ characters): 

▶ ERROR Bad passphrase: server rejected login attempt.
Please enter the Keybase passphrase for cameron.nicholson@fenixnet.net (12+ characters): 

The device you are currently using needs to be provisioned.
Which one of your existing devices would you like to use
to provision this new device?

	1. [computer]	laptop
	2. [computer]	ufomac
	3. [computer]	vagabond
	4. [paper key]	cause favorite
	5. I don't have access to any of these devices.

Choose a device: 3
Type this verification code into your other device:

	*****************************************

If you are using the command line client on your other device, run this command:

	keybase device add

It will then prompt you for the verification code above.

Enter a public name for this device: ufolaptop
▶ WARNING Account.CreateStreamCache overwriting exisitng StreamCache
✔ Success! You provisioned your device ufolaptop.

You are logged in as jadefalcon1
  - type `keybase help` for more info.

ufo:~ cameron$ keybase status

▶ WARNING device id mismatch session <-> env
Username:      jadefalcon1
Logged in:     no
Session:       jadefalcon1 [salt only]
    is valid:  no
    keys:      locked
    lksec:     loaded

KBFS:
    status:    not running
    version:   
    log:       /Users/cameron/Library/Logs/keybase.kbfs.log

Service:
    status:    running
    version:   1.0.15
    log:       /Users/cameron/Library/Logs/keybase.service.log

Platform Information:
    OS:        darwin
    Runtime:   go1.6
    Arch:      amd64

Client:
    version:   1.0.15

Desktop app:
    status:    not running
    version:   
    log:       /Users/cameron/Library/Logs/Keybase.app.log

Config path:   /Users/cameron/Library/Application Support/Keybase/config.json
Default user:  jadefalcon1
Other users:   

command-line client:
    keybase status [pid: 37969, version: 1.0.15]

ufo:~ cameron$ cat "/Users/cameron/Library/Application Support/Keybase/config.json"
{
    "current_user": "jadefalcon1",
    "users": {
        "jadefalcon1": {
            "device": "17018c9dbb99299e0d9abe6984e56818",
            "id": "ad9b4ad1764c46000f04943d8e884c19",
            "name": "jadefalcon1",
            "salt": "e203102c3527b262208e6e2614574b65"
        }
    }
}

**Comment by fenixnet-net** [comment]
I hope I didn't do it wrong...

**Comment by patrickxb** [comment]
No, I think I left out a crucial step:

1. `keybase ctl stop`
2. then replace the config.json file
3. `keybase ctl start`

Try that...

**Comment by fenixnet-net** [comment]
And now comes the part where I make your life even harder.

ufo:~ cameron$ keybase ctl stop
▶ INFO Stopping homebrew.mxcl.keybase
ufo:~ cameron$ vi Library/Application\ Support/Keybase/config.json 
ufo:~ cameron$ keybase ctl start
▶ ERROR No service (plist) installed with label: keybase.service

**Comment by patrickxb** [comment]
sheesh...homebrew...

`launchctl load ~/Library/LaunchAgents/homebrew.mxcl.keybase`

should do what `keybase ctl start` is trying to do...

@gabriel I think we might have a bug in `keybase ctl start` for homebrew installs.

**Comment by fenixnet-net** [comment]
You're gonna love this!

ufo:~ cameron$ launchctl load ~/Library/LaunchAgents/homebrew.mxcl.keybase
/Users/cameron/Library/LaunchAgents/homebrew.mxcl.keybase: No such file or directory
ufo:~ cameron$ find . -iname *keybase*
./Library/Application Support/Google/Chrome/Default/Local Storage/https_keybase.io_0.localstorage
./Library/Application Support/Google/Chrome/Default/Local Storage/https_keybase.io_0.localstorage-journal
./Library/Application Support/Keybase
./Library/Application Support/Keybase/keybase.leveldb
./Library/Caches/Keybase
./Library/Caches/Keybase/keybased.info
./Library/LaunchAgents/homebrew.mxcl.keybase.plist
./Library/Logs/homebrew.mxcl.keybase.log
./Library/Logs/keybase.service.log
./Library/Logs/keybase.start.log
ufo:~ cameron$ launchctl load ~/Library/LaunchAgents/homebrew.mxcl.keybase.plist 

That last one appears to have worked. However I'm still getting the scary mismatch error:
ufo:~ cameron$ keybase status
▶ WARNING device id mismatch session <-> env
Username:      jadefalcon1
Logged in:     no

Device:
    name:      laptop
    ID:        2dd0891fbc8a3318d13debe30261b718
    status:    active

Session:       no session
    is valid:  no
    keys:      locked
    lksec:     not loaded

KBFS:
    status:    not running
    version:   
    log:       /Users/cameron/Library/Logs/keybase.kbfs.log

Service:
    status:    running
    version:   1.0.15
    log:       /Users/cameron/Library/Logs/keybase.service.log

ufo:~ cameron$ keybase login
▶ WARNING device id mismatch session <-> env
ufo:~ cameron$ keybase status
Username:      jadefalcon1
Logged in:     yes

Device:
    name:      laptop
    ID:        2dd0891fbc8a3318d13debe30261b718
    status:    active

Session:       no session
    is valid:  yes
    keys:      locked
    lksec:     not loaded

KBFS:
    status:    not running
    version:   
    log:       /Users/cameron/Library/Logs/keybase.kbfs.log

Service:
    status:    running
    version:   1.0.15
    log:       /Users/cameron/Library/Logs/keybase.service.log

Platform Information:
    OS:        darwin
    Runtime:   go1.6
    Arch:      amd64

Client:
    version:   1.0.15

Desktop app:
    status:    not running
    version:   
    log:       /Users/cameron/Library/Logs/Keybase.app.log

Config path:   /Users/cameron/Library/Application Support/Keybase/config.json
Default user:  jadefalcon1
Other users:   

command-line client:
    keybase status [pid: 39241, version: 1.0.15]

**Comment by patrickxb** [comment]
Ok, but you only saw the scary error once, right?  Now you can do things like `keybase sign -m "it works!"`?

**Comment by fenixnet-net** [comment]
Correct. Now that I've logged in the scary message has gone away. Huzzah!

If you need any logs or other info I'll be happy to provide.

**Comment by patrickxb** [comment]
Ok, great!  No need for any further logs.  Thanks for your help!

**Comment by fenixnet-net** [comment]
And thank you very much for getting me into a working state.

**Comment by MichaelBlume** [comment]
Keybase seems to have deleted my /home/mike/.config/keybase/config.json and I can't log in without reprovisioning the device -- my issue seems to match californian's

**Comment by MichaelBlume** [comment]
fwiw, I have a ~/.config/keybase/secretkeys.michaelblume.mpack, just no config.json

also the paper key I thought I'd saved seems to have "not enough words", so it looks like I'll have to reset my account =/

**Comment by maxtaco** [comment]
you can recreate that file......

**Comment by maxtaco** [comment]
In the mean time, can you do a `keybase log send`?  We still haven't figured out who/what would be deleting your config, it's a big problem.

**Comment by maxtaco** [comment]
* Do a `keybase ctl stop`
* Put this file in `~/.config/keybase/config.json`:
```javascript
{ 
    "current_user": "michaelblume", 
    "users": { 
        "michaelblume": { 
            "device": "be20377774478699a724ab58533a9018", 
            "id": "fcccc4ada9f17c6c5385ced0b18ef519", 
            "name": "michaelblume", 
            "salt": "1a507c4a3c083e6a6e72d79f71ea99e8" 
      }
 }
 ```
* Run `keybase ctl start`

CC: @patrickxb  we should maybe make a utility to automate recreating these config files, especially if we can't get to the bottom of what in the world is deleting them.

**Comment by Californian** [comment]
In case this wasn't clear (it was a little buried in a multi-reply comment), I deleted my `~/.config` at one point (long story, but git versioning and symlinks saved the day), so it's possible that, if the `.mpack` file and the folder were recreated somehow, this was the cause, and it was not a keybase issue.

**Comment by MichaelBlume** [comment]
Done, 2e318757e2a4f76897e3971c

**Comment by MichaelBlume** [comment]
And that seems to have fixed it, thank you =)

**Comment by abliss** [comment]
FWIW, I found myself in this same position (and got here by googling). I found my `config.json` and `secretkeys.$USER.mpack` sitting in my home directory and there was no `.cache/keybase`. No idea how that happened (I'm running Ubuntu 14.04 and just update from 1.0.15 to 1.0.16 through `http://dist.keybase.io/linux/deb/repo/` ).

After creating the directory and moving those files into it, keybase seems to be working better now.
